pkg/recovery/repair.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package recovery
+
+import (
+	"context"
+
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/pss"
+	"github.com/ethersphere/bee/pkg/pushsync"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/bee/pkg/tags"
+	"github.com/ethersphere/bee/pkg/trojan"
+)
+
+const (
+	// RecoveryTopicText is the string used to construct the recovery topic.
+	RecoveryTopicText = "RECOVERY"
+)
+
+var (
+	// RecoveryTopic is the topic used for repairing globally pinned chunks.
+	RecoveryTopic = trojan.NewTopic(RecoveryTopicText)
+)
+
+// RecoveryHook defines code to be executed upon failing to retrieve chunks.
+type RecoveryHook func(chunkAddress swarm.Address, targets trojan.Targets) error
+
+// sender is the function call for sending trojan chunks.
+type PssSender interface {
+	Send(ctx context.Context, targets trojan.Targets, topic trojan.Topic, payload []byte) (*tags.Tag, error)
+}
+
+// NewRecoveryHook returns a new RecoveryHook with the sender function defined.
+func NewRecoveryHook(pss PssSender) RecoveryHook {
+	return func(chunkAddress swarm.Address, targets trojan.Targets) error {
+		payload := chunkAddress
+		ctx := context.Background()
```

Comment:
Would it be more useful for RecoveryHook to accept context instead to create a new one here?

Reply:
i changed like this because of this review comment https://github.com/ethersphere/bee/pull/479#discussion_r466899136.

Reply:
I am aware. I am sorry, I assumed that the reason for context propagation is cleat.